### PR TITLE
fix: support reimburse quota if download consumed extra quota

### PR DESCRIPTION
**Pull request by flywukong**
### Description

fix quota consumed method. Use the actual downloaded size during download processing to calculate the value of the consumed quota,  if the extra quota has been consumed , the getObject handler will make a request to downloader to fix the quota traffic db

### Rationale

fix the download quota consumed method

### Example
NA

### Changes

Notable changes: 
* fix the download quota

### CODE REVIEW

File: store/sqldb/traffic.go [hunk 1]
```diff
@@ -258,6 +258,91 @@ func (s *SpDBImpl) GetBucketTraffic(bucketID uint64, yearMonth string) (traffic
 	}, nil
 }
 
+// UpdateExtraQuota update the read consumed quota and free consumed quota in traffic db with the extra quota
+func (s *SpDBImpl) UpdateExtraQuota(bucketID, extraQuota uint64) error {
```
Comment (sysvm) [inline store/sqldb/traffic.go:262]: Please add UT.

File: modular/downloader/download_task.go [hunk 2]
```diff
@@ -279,17 +279,6 @@ func (d *DownloadModular) PreDownloadPiece(ctx context.Context, downloadPieceTas
 		bucketID := downloadPieceTask.GetBucketInfo().Id.Uint64()
 		bucketName := downloadPieceTask.GetBucketInfo().GetBucketName()
 
-		// if stream account has been frozen, it is not allowed to download
```
Comment (mliu233) [inline modular/downloader/download_task.go:282]: Is the below logic no longer valid?

Comment (flywukong) [reply modular/downloader/download_task.go:282]: no, if it is getObject handler , it is already checked in func (a *AuthenticationModular) VerifyAuthentication() . So This is a duplicate check

**Review by annielz — APPROVED** [review]

**Review by sysvm — APPROVED** [review]
LGTM

**Pull request merged**